include/SDL3/SDL_platform_defines.h

```diff
@@ -65,7 +65,21 @@
 #define SDL_PLATFORM_APPLE  1
 /* lets us know what version of macOS we're compiling on */
 #include <AvailabilityMacros.h>
+/* macOS 14 SDK has a problematic use of __has_extension in TargetConditionals.h
+   https://github.com/libsdl-org/SDL/issues/9632 */
+#ifndef __has_extension
+# ifdef __has_feature
+#  define __has_extension __has_feature
+# else
+#  define __has_extension(x) 0
+# endif
+# define SDL_undef_has_extension
+#endif /* __has_extension */
```

Comment:
I think it makes more sense to just include TargetConditionals.h in both branches rather than define a separate define to undefine.

Reply:
Here's a proposed patch
```diff
diff --git a/include/SDL3/SDL_platform_defines.h b/include/SDL3/SDL_platform_defines.h
index 72b1ef260..b418ca411 100644
--- a/include/SDL3/SDL_platform_defines.h
+++ b/include/SDL3/SDL_platform_defines.h
@@ -65,7 +65,17 @@
 #define SDL_PLATFORM_APPLE  1
 /* lets us know what version of macOS we're compiling on */
 #include <AvailabilityMacros.h>
+/* The macOS 14 SDK uses __has_extension in TargetConditionals.h */
+#ifndef __has_extension
+# ifdef __has_feature
+#  define __has_extension __has_feature
+# else
+#  define __has_extension(x) 0
+# endif
 #include <TargetConditionals.h>
+#else
+#include <TargetConditionals.h>
+#endif

 /* Fix building with older SDKs that don't define these
    See this for more information:

```

Reply:
Are we sure that __has_extension and __has_feature are equivalent? Have we heard back from the original bug reporter to find out what compiler they're using?

Reply:
> I think it makes more sense to just include TargetConditionals.h in both branches rather than define a separate define to undefine.

The undefine was suggested by @icculus in #9632 (and I kind'a agree with it)

Reply:
> Are we sure that __has_extension and __has_feature are equivalent?

They aren't exactly the same, but, as I said, that's what clang docs seem to suggest to use if __has_extension is absent: https://clang.llvm.org/docs/LanguageExtensions.html (search __has_extension on that page)

> Have we heard back from the original bug reporter to find out what compiler they're using?

Nope. That's why I CC'd him here

Reply:
>  find out what compiler they're using?

However, I do think they're using a version of gcc: all versions of clang I tried do have __has_extension. As for gcc, gcc-10 doesn't have either of __has_extension and __has_feature, however gcc-14 has both of them.

Reply:
Rebased and simplified the patch a bit.